inc/c_util/interlocked_hl.h

```diff
@@ -22,6 +22,7 @@ extern "C" {
 #define INTERLOCKED_HL_RESULT_VALUES \
     INTERLOCKED_HL_OK, \
     INTERLOCKED_HL_ERROR, \
+    INTERLOCKED_HL_TIMEOUT, \
     INTERLOCKED_HL_CHANGED
```

Comment:
Consider a task to create a separate enum for InterlockedHL_WaitForValue (and likely InterlockedHL_WaitForNotValue)